SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt

```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
 
-                    // -- Focus
-                    val focusManager = LocalFocusManager.current
-
                     // -- Tabs
                     val tabs = stringArrayResource(id = R.array.trade_flow_tabs)
+                    val selectedTabIndex = remember { mutableStateOf(0) }
 
-                    // -- Value Input Type
+                    // -- Currency input
                     val currencyState = remember { mutableStateOf(asset) }
-                    val currentValueInput = remember { mutableStateOf(AssetBankModel.Type.fiat) }
-                    val valueInput = remember { mutableStateOf("") }
-                    val valueLabelHintAsset = if (currentValueInput.value == AssetBankModel.Type.fiat) pairAsset else currencyState.value
-                    val amountHint = buildAnnotatedString {
-                        append(stringResource(id = R.string.trade_flow_text_field_amount_placeholder))
-                        withStyle(style = SpanStyle(
-                            color = colorResource(id = R.color.list_prices_asset_component_code_color))) {
-                            append(" ${valueLabelHintAsset.code}")
-                        }
-                    }
-                    val amountLabel = buildAnnotatedString {
-                        append(stringResource(id = R.string.trade_flow_text_field_amount_placeholder))
-                        append(" ${valueLabelHintAsset.code}")
-                    }
 
-                    // -- DropDown
-                    val expanded = remember { mutableStateOf(false) }
-                    val textFieldSize = remember { mutableStateOf(Size.Zero) }
-                    val selectedTabIndex = remember { mutableStateOf(0) }
-                    val icon = Icons.Filled.ArrowDropDown
-
-                    TabRow(
-                        selectedTabIndex = selectedTabIndex.value,
-                        backgroundColor = Color.Transparent,
-                        indicator = { tabsIndicators ->
-                            Box(
-                                Modifier
-                                    .tabIndicatorOffset(tabsIndicators[selectedTabIndex.value])
-                                    .height(2.dp)
-                                    .border(3.5.dp, colorResource(id = R.color.primary_color))
+                    // -- Currency Input --> DropDown
+                    val expandedCurrencyInput = remember { mutableStateOf(false) }
+                    val currencyInputWidth = remember { mutableStateOf(Size.Zero) }
+
+                    // -- Value Input
+                    val typeOfValueState = remember { mutableStateOf(AssetBankModel.Type.fiat) }
+                    val valueState = remember { mutableStateOf("") }
+                    val valueAsset = remember { mutableStateOf(
+                        if (typeOfValueState.value == AssetBankModel.Type.fiat)
+                            pairAsset else currencyState.value
+                    )}
```

Comment:
What's going on here?

Reply:
We have the MutableState of AssentBankModel, but to assign the first value, we have this if that check the "typeOfValueState" (check if the input its accepting fiat or crypto) and depend of this to assign this model to get "code" and "name" for lables and placeholders